### PR TITLE
feat: a bunch of tests are passing now

### PR DESCRIPTION
**Pull request by kentcdodds**
ref #2

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
-  marginRight: 'marginLeft',
+// this will be an object of properties that map to their corresponding rtl property (their doppelganger)
+const propertiesToConvert = arrayToObject([
+  ['paddingLeft', 'paddingRight'],
+  ['marginLeft', 'marginRight'],
+  ['left', 'right'],
+])
+
+// this is the same as the propertiesToConvert except for values
+const valuesToConvert = arrayToObject([
```
Comment (xymostech) [inline src/index.js]: Should these values be scoped to specific properties? I guess I don't know if 'left' would ever be a valid value of some property that we wouldn't want to switch...

File: src/index.js [hunk 2]
```diff
@@ -1,12 +1,55 @@
...
+    }
+    const [top, right, bottom, left, important] = splitValues
+    if (important && important !== '!important') {
+      return value // the value is invalid and we can't do anything about that...
+    }
+    return [top, left, bottom, right, important].filter(Boolean).join(' ')
+  },
+  textShadow(value) {
+    // intentionally leaving off the `g` flag here because we only want to change the first number (which is the offset-x)
+    return value.replace(/(-*)(\d+)/, (match, negative, number) => {
```
Comment (xymostech) [inline src/index.js]: Handy that all of the different values for text shadow have this property!

File: src/index.js [hunk 3]
```diff
@@ -15,11 +58,71 @@ function convert(object) {
...
+ * @param {String|Number|Object} originalValue the original css property value. If it's an object, then we'll convert that as well
+ * @return {String|Number|Object} the converted value
+ */
+function getValueDoppelganger(key, originalValue) {
+  /* eslint complexity:[2, 7] */ // let's try to keep the complexity down... If we have to do this much more, let's break this up
+  if (isObject(originalValue)) {
+    return convert(originalValue) // recurssion 🌀
+  }
+  const isNum = isNumber(originalValue)
+  const importantlessValue = isNum ? originalValue : originalValue.replace(/ !important.*?$/, '')
```
Comment (xymostech) [inline src/index.js]: If you're handling !important here, you probably don't need to handle it in `padding`, right?

**Comment by xymostech** [comment]
I love your code coverage bot too!

File: src/index.js (hunk 3 above)
Comment (kentcdodds) [inline src/index.js]: True, I'll go ahead and merge as is because I'm on my phone. We can optimize later

**Pull request merged**